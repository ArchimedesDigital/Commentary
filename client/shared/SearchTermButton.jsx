import FlatButton from 'material-ui/FlatButton';
import FontIcon from 'material-ui/FontIcon';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

SearchTermButton = React.createClass({

	propTypes: {
		toggleSearchTerm: React.PropTypes.func.isRequired,
		label: React.PropTypes.string.isRequired,
		searchTermKey: React.PropTypes.string.isRequired,
		value: React.PropTypes.object.isRequired,
		activeWork: React.PropTypes.bool,
		active: React.PropTypes.bool,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},

	toggleSearchTerm() {
		this.props.toggleSearchTerm(this.props.searchTermKey, this.props.value);
	},

	render() {
		const styles = {
			iconStyle: {
				color: '#999',
			},
		};
		let className = 'search-term-button';

		if (this.props.active || this.props.activeWork) {
			className += ' search-term-button--active';
		}

		return (
			<li>
				<FlatButton
					className={className}
					onClick={this.toggleSearchTerm}
					label={this.props.label}
					icon={
						<FontIcon
							className="mdi mdi-plus-circle-outline"
							style={styles.iconStyle}
						/>
					}
				/>
			</li>
		);
	},
});

export default SearchTermButton;
