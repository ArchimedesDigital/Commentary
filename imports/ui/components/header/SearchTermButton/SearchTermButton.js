import React from 'react';
import FontIcon from 'material-ui/FontIcon';

/*
	helpers
*/
const getClassName = (active, activeWork) => {
	let className = 'search-term-button';

	if (active || activeWork) {
		className += ' search-term-button--active';
	}

	return className;
};


const SearchTermButton = ({ toggleSearchTerm, label, searchTermKey, value, activeWork, active }) => (
	<li>
		<button
			className={getClassName(active, activeWork)}
			onClick={toggleSearchTerm.bind(null, searchTermKey, value)}
		>
			<span className="search-term-button-label">
				{label}
			</span>
			<FontIcon
				className="mdi mdi-plus-circle-outline search-term-button-icon"
			/>
		</button>
	</li>
);
SearchTermButton.propTypes = {
	toggleSearchTerm: React.PropTypes.func.isRequired,
	label: React.PropTypes.string.isRequired,
	searchTermKey: React.PropTypes.string.isRequired,
	value: React.PropTypes.any.isRequired, // eslint-disable-line react/forbid-prop-types
	activeWork: React.PropTypes.bool,
	active: React.PropTypes.bool,
};
SearchTermButton.defaultProps = {
	activeWork: false,
	active: false,
};

export default SearchTermButton;
