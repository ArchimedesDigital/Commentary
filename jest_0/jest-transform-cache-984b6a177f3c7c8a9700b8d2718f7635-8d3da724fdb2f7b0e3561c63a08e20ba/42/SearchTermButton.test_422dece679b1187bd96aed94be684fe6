/* istanbul ignore next */'use strict';


var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_chai = require('chai'); /* istanbul ignore next */var _chai2 = _interopRequireDefault(_chai);
var /* istanbul ignore next */_sinon = require('sinon'); /* istanbul ignore next */var _sinon2 = _interopRequireDefault(_sinon);


var /* istanbul ignore next */_SearchTermButton = require('./SearchTermButton'); /* istanbul ignore next */var _SearchTermButton2 = _interopRequireDefault(_SearchTermButton);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('SearchTermButton', function () {
	it('should render', function () {
		var toggleSearchTerm = /* istanbul ignore next */_sinon2.default.spy();
		var wrapper = /* istanbul ignore next */(0, _enzyme.shallow)(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_SearchTermButton2.default, /* istanbul ignore next */{
			toggleSearchTerm: toggleSearchTerm,
			label: 'TestSearchterm',
			searchTermKey: 'TestKey',
			value: { key: 'value' } }));


		var button = wrapper.find('button');
		/* istanbul ignore next */_chai2.default.assert.ok(button, 'should contain the button child');
	});

	it('should render with active prop set', function () {
		var toggleSearchTerm = /* istanbul ignore next */_sinon2.default.spy();
		var wrapper = /* istanbul ignore next */(0, _enzyme.shallow)(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_SearchTermButton2.default, /* istanbul ignore next */{
			toggleSearchTerm: toggleSearchTerm,
			label: 'TestSearchterm',
			searchTermKey: 'TestKey',
			value: { key: 'value' },
			active: /* istanbul ignore next */true }));


		var button = wrapper.find('button');
		var props = wrapper.instance().props;
		/* istanbul ignore next */_chai2.default.assert.ok(button, 'should contain the button child');
		/* istanbul ignore next */_chai2.default.assert.equal(props.label, 'TestSearchterm');
		/* istanbul ignore next */_chai2.default.assert.include(button.node.props.className, 'search-term-button--active');
	});

	it('should render with activeWork prop set', function () {
		var toggleSearchTerm = /* istanbul ignore next */_sinon2.default.spy();
		var wrapper = /* istanbul ignore next */(0, _enzyme.shallow)(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_SearchTermButton2.default, /* istanbul ignore next */{
			toggleSearchTerm: toggleSearchTerm,
			label: 'TestSearchterm',
			searchTermKey: 'TestKey',
			value: { key: 'value' },
			activeWork: /* istanbul ignore next */true }));


		var button = wrapper.find('button');
		var props = wrapper.instance().props;
		/* istanbul ignore next */_chai2.default.assert.ok(button, 'should contain the button child');
		/* istanbul ignore next */_chai2.default.assert.equal(props.label, 'TestSearchterm');
		/* istanbul ignore next */_chai2.default.assert.include(button.node.props.className, 'search-term-button--active');
	});

	it('should call parent callback with correct values', function () {
		var toggleSearchTerm = /* istanbul ignore next */_sinon2.default.spy();
		var wrapper = /* istanbul ignore next */(0, _enzyme.shallow)(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_SearchTermButton2.default, /* istanbul ignore next */{
			toggleSearchTerm: toggleSearchTerm,
			label: 'TestSearchterm',
			searchTermKey: 'TestKey',
			value: { key: 'value' } }));


		var button = wrapper.find('button');
		var props = wrapper.instance().props;
		button.simulate('click');
		/* istanbul ignore next */_sinon2.default.assert.calledWith(toggleSearchTerm, 'TestKey', { key: 'value' });
		/* istanbul ignore next */_chai2.default.assert.ok(button, 'should contain the button child');
		/* istanbul ignore next */_chai2.default.assert.equal(props.label, 'TestSearchterm');
	});
}); // component:
/* eslint-env mocha */ /* eslint-disable func-names, prefer-arrow-callback, import/no-extraneous-dependencies */