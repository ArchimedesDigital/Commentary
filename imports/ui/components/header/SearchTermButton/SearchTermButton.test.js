/* eslint-env mocha */
/* eslint-disable func-names, prefer-arrow-callback, import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';

// component:
import SearchTermButton from './SearchTermButton'; 

describe('SearchTermButton', () => {
	it('should render', () => {
		const toggleSearchTerm = sinon.spy();
		const wrapper = shallow(
			<SearchTermButton
				toggleSearchTerm={toggleSearchTerm}
				label="TestSearchterm"
				searchTermKey="TestKey"
				value={{ key: 'value' }}
			/>
		);
		const button = wrapper.find('button');
		chai.assert.ok(button, 'should contain the button child');
	});

	it('should render with active prop set', () => {
		const toggleSearchTerm = sinon.spy();
		const wrapper = shallow(
			<SearchTermButton
				toggleSearchTerm={toggleSearchTerm}
				label="TestSearchterm"
				searchTermKey="TestKey"
				value={{ key: 'value' }}
				active
			/>
		);
		const button = wrapper.find('button');
		const props = wrapper.instance().props;
		chai.assert.ok(button, 'should contain the button child');
		chai.assert.equal(props.label, 'TestSearchterm');
		chai.assert.include(button.node.props.className, 'search-term-button--active');
	});

	it('should render with activeWork prop set', () => {
		const toggleSearchTerm = sinon.spy();
		const wrapper = shallow(
			<SearchTermButton
				toggleSearchTerm={toggleSearchTerm}
				label="TestSearchterm"
				searchTermKey="TestKey"
				value={{ key: 'value' }}
				activeWork
			/>
		);
		const button = wrapper.find('button');
		const props = wrapper.instance().props;
		chai.assert.ok(button, 'should contain the button child');
		chai.assert.equal(props.label, 'TestSearchterm');
		chai.assert.include(button.node.props.className, 'search-term-button--active');
	});

	it('should call parent callback with correct values', () => {
		const toggleSearchTerm = sinon.spy();
		const wrapper = shallow(
			<SearchTermButton
				toggleSearchTerm={toggleSearchTerm}
				label="TestSearchterm"
				searchTermKey="TestKey"
				value={{ key: 'value' }}
			/>
		);
		const button = wrapper.find('button');
		const props = wrapper.instance().props;
		button.simulate('click');
		sinon.assert.calledWith(toggleSearchTerm, 'TestKey', { key: 'value' });
		chai.assert.ok(button, 'should contain the button child');
		chai.assert.equal(props.label, 'TestSearchterm');
	});
});
