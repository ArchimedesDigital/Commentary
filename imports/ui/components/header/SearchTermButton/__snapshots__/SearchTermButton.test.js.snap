// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTermButton renders correctly 1`] = `
<li>
  <button
    className="search-term-button"
    onClick={[Function]}
  >
    <span
      className="search-term-button-label"
    >
      TestSearchterm
    </span>
    <span
      className="mdi mdi-plus-circle-outline search-term-button-icon"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    />
  </button>
</li>
`;
